go/cloudbuild-test.yaml

```diff
@@ -50,3 +50,20 @@ steps:
         "-v",
       ]
     waitFor: ["go123-build"]
+
+  # Go 1.23 release build
+  - name: gcr.io/cloud-builders/docker
+    args: ["build", "-t", "gcr.io/$PROJECT_ID/release-images/go123", "."]
```

Comment:
The prefix for the container destination still does not match.

Reply:
Should it? Is `us-central1-docker.pkg.dev` a replacement for `gcr.io` in all cases? Happy to make the change, I'm just not aware of what is correct.

Reply:
/cc @chingor13

Reply:
I believe the use of gcr.io is deprecated: https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

Reply:
In this case it doesn't matter for the test as these aren't actually pushed to AR or CR.

The us-central1-docker.pkg.dev is suggested moving forward though.

Reply:
Changing `gcr.io` to `us-central1-docker.pkg.dev` throughout the entire file.